From ba7101b681f95bf8106513e2a0ffe1622f816ff0 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Wed, 19 Nov 2025 16:24:02 +0100
Subject: [PATCH 10/12] Revert "xorg: initialize glamor provider"

This reverts commit 0a1ee643b26fd53074ec90c95a14c12c8f5166b3.

This is causing a number of regressions on existing setups:

 * Reverse PRIME with the NVIDIA proprietary driver, where software
   rendering is used instead of the NVIDIA GLX library with hardware
   acceleration
 * Performance issues with AMDGPU
 * Rendering with 10-bit output with AMDGPU

Revert the change that is causing these regressions in the stable branch.

Closes: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1848
Part-of: <https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/2104>
---
 glamor/glamor_egl.c               | 11 -----------
 hw/xfree86/glamor_egl/meson.build |  2 +-
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/glamor/glamor_egl.c b/glamor/glamor_egl.c
index a3ea4c3d5..dc31c1650 100644
--- a/glamor/glamor_egl.c
+++ b/glamor/glamor_egl.c
@@ -46,7 +46,6 @@
 
 #include "glamor.h"
 #include "glamor_priv.h"
-#include "glamor_glx_provider.h"
 #include "dri3.h"
 
 struct glamor_egl_screen_private {
@@ -896,9 +895,6 @@ glamor_egl_screen_init(ScreenPtr screen, struct glamor_context *glamor_ctx)
         glamor_egl_get_screen_private(scrn);
 #ifdef DRI3
     glamor_screen_private *glamor_priv = glamor_get_screen_private(screen);
-#endif
-#ifdef GLXEXT
-    static Bool vendor_initialized = FALSE;
 #endif
     const char *gbm_backend_name;
 
@@ -940,13 +936,6 @@ glamor_egl_screen_init(ScreenPtr screen, struct glamor_context *glamor_ctx)
         }
     }
 #endif
-#ifdef GLXEXT
-    if (!vendor_initialized) {
-        GlxPushProvider(&glamor_provider);
-        xorgGlxCreateVendor();
-        vendor_initialized = TRUE;
-    }
-#endif
 }
 
 static void glamor_egl_cleanup(struct glamor_egl_screen_private *glamor_egl)
diff --git a/hw/xfree86/glamor_egl/meson.build b/hw/xfree86/glamor_egl/meson.build
index dd1cafcd9..7eae05812 100644
--- a/hw/xfree86/glamor_egl/meson.build
+++ b/hw/xfree86/glamor_egl/meson.build
@@ -15,7 +15,7 @@ shared_module(
         dependency('libdrm', version: '>= 2.4.46'),
         gbm_dep,
     ],
-    link_with: [glamor, libxserver_glx],
+    link_with: glamor,
 
     install: true,
     install_dir: module_dir,
-- 
2.52.0

